include/envoy/filesystem/filesystem.h

```diff
@@ -61,5 +61,30 @@ class Watcher {
 
 typedef std::unique_ptr<Watcher> WatcherPtr;
 
+/**
+ * Abstraction for listing a directory.
+ * TODO(sesmith177): replace with std::filesystem::directory_iterator once we move to C++17
```

Comment:
Is there any way to use `std::experimental::filesystem` in the interim as well? I know we've hit issues on Mac OS X because of this in the past when trying, see https://github.com/envoyproxy/envoy/blob/master/test/test_common/environment.cc#L35 for example.

Reply:
I'm not sure I understand -- I assumed we wouldn't want to use `std::experimental::filesystem` in non-test code?

Reply:
I think the API is experimental, not the implementation, but I could be mistaken. In any case, it's probably not sufficiently cross-platform today.